src/networkapi.jl

```diff
@@ -603,19 +603,19 @@ For example, continuing the example from [`deficiency`](@ref)
    subnets = subnetworks(sir, linkage_classes)
 ```
 """
-function subnetworks(rs::ReactionSystem, lcs::Vector{Vector{Int64}};
+function subnetworks(rs::ReactionSystem, lcs::AbstractVector;
                   rxs = reactions(rs),
-                  complextorxmap = collect(values(reactioncomplexmap(rs))),
+                  complextorxmap = [map(first,rcmap) for rcmap in values(reactioncomplexmap(rs))],
                   p = parameters(rs))
 
-   t = ModelingToolkit.get_iv(rs)
-   subreac = Vector{ReactionSystem}()
-   for i in 1:length(lcs)
-      reacs, specs, newps = subnetworkmapping(lcs[i], rxs, complextorxmap[lcs[i]] ,p)
-      newname = Symbol(nameof(rs), "_", i)
-      push!(subreac , ReactionSystem(reacs, t, specs, newps;name = newname))
-   end
-   subreac
+    t = get_iv(rs)
+    subreac = Vector{ReactionSystem}()
+    for i in 1:length(lcs)
+        reacs, specs, newps = subnetworkmapping(lcs[i], rxs, complextorxmap, p)
```

Comment:
shouldn't we just pass `complextorxmap[lcs[i]]` ? i.e. reaction indices corresponding to complexes in linkageclass `lcs[i]`

Reply:
> shouldn't we just pass `complextorxmap[lcs[i]]` ? i.e. reaction indices corresponding to complexes in linkageclass `lcs[i]`

I take this back! your way is efficient, just checked with `@time` n `@btime`.

Reply:
Yeah, I was trying to avoid allocations of intermediary vectors.